service/grails-app/controllers/mod/rs/SettingController.groovy

```diff
@@ -114,28 +117,47 @@ class SettingController extends OkapiTenantAwareSwaggerController<AppSetting> {
         ContextLogging.setValue(ContextLogging.FIELD_ACTION, ContextLogging.ACTION_SEARCH);
         log.debug(ContextLogging.MESSAGE_ENTERING);
 
-        // Use gorm criteria builder to always add your custom filter....
-        Closure gormFilterClosure = {
-            or {
-                isNull('hidden')
-                eq('hidden', false)
-            }
-        };
-
-        // Are they explicitly filtering on the hidden field
-        if (params.filters != null) {
-            // they are, so see if hidden is being filtered on
-            if (params.filters.toString().indexOf("hidden") > -1) {
-                // They are explicitly filtering on it, so we do want to return hidden settings
-                gormFilterClosure = null;
+        def result = []
+
+        AppSetting.withNewSession { session ->
+            AppSetting.withNewTransaction { status ->
+                // Use GORM criteria builder to add your custom filter....
+                Closure gormFilterClosure = {
+                    or {
+                        isNull('hidden')
+                        eq('hidden', false)
+                    }
+                }
+
+                // Check if explicitly filtering on the hidden field
+                if (params.filters != null && params.filters.toString().indexOf("hidden") > -1) {
+                    // They are explicitly filtering on it, so use null to avoid altering
+                    gormFilterClosure = null
+                }
+
+                // Perform the lookup
+                result = doTheLookup(gormFilterClosure)
+
+                // Determine if we need to filter based on the feature flag
+                if (params.filters.contains(SettingsData.SECTION_STATE_ACTION_CONFIG)) {
+                    // Iterate through each record in the result
+                    result = result.findAll { record ->
+                        // Construct the feature flag value for state action configuration
+                        String featFlagKey = record.section + "." + record.key + "." + "feature_flag"
+                        String featFlagValue = settingsService.getSettingValue(featFlagKey)
+
+                        // Filter only if the featureFlag is not null and equals "false"
+                        !(featFlagValue != null && featFlagValue == "false" && record.key == SettingsData.SETTING_COMBINE_FILL_AND_SHIP)
```

Comment:
This line is wrong. Remove the hardcoded SettingsData. Was this tested on any other setting?